### DISCUSSION TITLE
FT810 and ESP32-S3-DevKitC-1 how to use

### DISCUSSION THREAD

**Discussion by ajan0636** [RudolphRiedel/FT800-FT813#145] [Q&A]
Hi, good day

First of all, thank you for the library. I have an "**ESP32-S3-DevKitC-1**" but I can't seem to build it, I always get errors. I am using "**PlatformIO**" wtih VSCode

Tx for helping

**Comment by RudolphRiedel** [comment]
Hmm, my PlatformIO just killed itself, again, "MissingPackageManifestError: Could not find one of 'package.json' manifest files in the package".

But before it did I successfully added board = esp32-s3-devkitc-1 to both the Arduino example in this repository and to the version I have on  my drive.
And all three variants build now after I manually deleted a couple of ESP32 packages from the .platformio folder.
I also changed the display to one using a FT813 to make sure there is nothing for BT81x there by accident.

Other than a gazillion of warnings for the Arduino sources and a number of warnings for the ESP-IDF headers, this seems to be going fine.

So I guess I need you to be more specific, what errors, and what does your platformio.ini look like?

**Comment by ajan0636** [comment]
Yes, I have fixed the errors, successfully added the library, and the Hello World example was successfully sent to the processor. However, nothing is displayed on the screen. I am using the FT810 and I am sure I have set the pins correctly, as they are in the platformio.ini file. Where could I be going wrong?


```

; PlatformIO Project Configuration File
;
;   Build options: build flags, source filter
;   Upload options: custom upload port, speed and extra flags
;   Library options: dependencies, extra library storages
;   Advanced options: extra scripting
;
; Please visit documentation for the other options and examples
; https://docs.platformio.org/page/projectconf.html

[env]
lib_deps = https://github.com/RudolphRiedel/FT800-FT813/archive/refs/heads/5.x.zip
framework = arduino
lib_ldf_mode = chain+
build_flags = -D EVE_RVT50H

[env:esp32-s3-devkitc-1]
platform = espressif32
board = esp32-s3-devkitc-1
framework = arduino
build_unflags = -Os
build_flags =
    ${env.build_flags}
    -D EVE_USE_ESP_IDF
    -D EVE_CS=10
    -D EVE_PDN=14
    -D EVE_SCK=12
    -D EVE_MISO=13
    -D EVE_MOSI=11
    -O2

```

**Reply by RudolphRiedel** [reply]
Which display are you using?
> build_flags = -D EVE_RVT50H

This is for a module from Riverdi with a BT817 and 800x480 which makes the library send commands that a FT810 can not understand.

**Reply by ajan0636** [reply]
4.3" 480X272 TFT DISPLAY

**Reply by ajan0636** [reply]
yes u are right now am restore like this 

`build_flags = -D EVE_FT810CB_HY50HD`

but still not working

**Reply by RudolphRiedel** [reply]
The EVE_FT810CB_HY50HD is 800x480.
The only FT810 based module with 480x272 a have a config for would be EVE_GAMEDUINO3.

Is this a ready-to-use module or are you using your own PCB with a FT810?

It might be necessary to have a look with a logic analyser to see if the FT810 is even responding.

**Comment by ajan0636** [comment]
its my own pcb i created ft810 and 4.3" tft

**Reply by RudolphRiedel** [reply]
I have given up on that idea quite some time ago. :-)
Ok now, which panel are you using?
Does your FT810 have a crystal or not?

**Comment by ajan0636** [comment]
panel = https://www.hotmcu.com/43-inch-480x272-tft-lcd-display-touch-panel-standard-40-pin-p-43.html?cPath=6_16
datasheet = https://www.haoyuelectronics.com/Attachment/HY43-LCD/LCD%20DataSheet.pdf

yes i have 12MHz crystal

I'm new to ESP32 and don't have much software knowledge, I'm sorry. I entered the technical specifications of the panel, but I still couldn't get a result.

i rewrote for my panel:

EVE_config.h /932 line

```
#if defined (EVE_FT810CB_HY50HD)
#define EVE_HSIZE (480)
#define EVE_VSIZE (272)

#define EVE_VSYNC0 (0)
#define EVE_VSYNC1 (10)
#define EVE_VOFFSET (12)
#define EVE_VCYCLE (292)
#define EVE_HSYNC0 (0)
#define EVE_HSYNC1 (41)
#define EVE_HOFFSET (43)
#define EVE_HCYCLE (548)
#define EVE_PCLK (2)
#define EVE_PCLKPOL (1)
#define EVE_SWIZZLE (0)
#define EVE_CSPREAD (1)
#define EVE_HAS_CRYSTAL
#define EVE_GEN 2
#endif

```

**Reply by RudolphRiedel** [reply]
Oh, ok, HOTMCU / HAOYU, it really has been a while since I used their modules. :-)
They do have a FT800CB-HY43B - which is not supported by V5 of this library.
And they have the FT810CB-HY43B-HD.

I am not using resistive panels anymore though.

Your configuration values almost look plausible, except for "#define EVE_PCLK (2)".
That would set the pixel clock to 30MHz, the panel can not work with that.
A value of 6 would be optimal given the rest of the parameters.

The datasheet has slightly different values for the rest like VCYCLE, but generally these values are not as critical.
